backend/src/cusco/sources/entities.py

```
@@ -37,40 +40,45 @@ class EntitiesSource(DataSource):
+            logger.info(
+                f"IMPIC entities: SQLite fresh — {row_count} entities "
+                "(skipping download)"
+            )
+            self._loaded = True
             return
+
         try:
             data = await self._load_entities()
-            self._index_entities(data)
+            await self._persist_entities(data)
         except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)